xtask/src/build_docker_image.rs

```diff
@@ -187,7 +187,7 @@ pub fn build_docker_image(
 
         if push {
             docker_build.arg("--push");
-        } else if engine.kind.is_docker() && no_output {
+        } else if engine.kind.supports_output_flag() && no_output {
```

Comment:
I feel like this should be an error, i.e it should not be possible to specify no output and then not have it happen

Reply:
I've changed it so we error if it doesn't support the output flag, with another branch:

```rust
if push {
    docker_build.arg("--push");
} else if engine.kind.supports_output_flag() && no_output {
    docker_build.args(["--output", "type=tar,dest=/dev/null"]);
} else if no_output {
    msg_info.fatal("cannot specify `--no-output` with engine that does not support the `--output` flag", 1);
} else if has_buildkit {
    docker_build.arg("--load");
}
```